runtime/sharedstate_test.go

```diff
@@ -206,86 +205,204 @@ func TestRuntimeSharedState(t *testing.T) {
 			Interface:   runtimeInterface,
 			Location:    nextTransactionLocation(),
 			Environment: environment,
+			UseVM:       *compile,
 		},
 	)
 	require.NoError(t, err)
 
 	assert.Equal(t, []string{`"Hello from C2!"`}, loggedMessages)
 
-	// Assert shared state was used,
-	// i.e. data was not re-read
+	if *compile {
 
-	expectedReads := []ownerKeyPair{
-		// Read account register to check if it is a migrated account
-		// Read returns no value.
-		{
-			owner: signerAddress[:],
-			key:   []byte(AccountStorageKey),
-		},
-		// Read contract domain register.
-		// Read returns no value.
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainContract.Identifier()),
-		},
-		// Read all available domain registers to check if it is a new account
-		// Read returns no value.
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.PathDomainStorage.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.PathDomainPrivate.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.PathDomainPublic.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainContract.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainInbox.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainCapabilityController.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainCapabilityControllerTag.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainPathCapability.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(common.StorageDomainAccountCapability.Identifier()),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(AccountStorageKey),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(AccountStorageKey),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte(AccountStorageKey),
-		},
-		{
-			owner: signerAddress[:],
-			key:   []byte{'$', 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2},
-		},
-	}
+		// Shared state is ignored in VM environment
```

Comment:
Does this mean we produce different state-changes in VM? or is this something else?

Reply:
Opened https://github.com/onflow/cadence/pull/4078